portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,9 +32,27 @@
   @Override
 
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+      // store user comment 
       String userComment = getComment(request);
       ArrayList<String> commentArray = new ArrayList<String>();
       commentArray.add(userComment);
+      System.out.println(commentArray);
+
+      // get entity properties
+      String title = request.getParameter("title");
```

Comment:
https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)
getParameter returns null if the param doesn't exits, so this line is fine, but taskEntity.setProperty("title", title); might have some issues. What happens if we pass a null value here? This is an example of something that we would want to test, but considering we haven't done testing yet, I would encourage you to try out literally passing a null value to setProperty. If passing through null doesn't cause any unexpected behavior, then we're fine. But if it causes unexpected behavior, we might want to test for null before creating an entity. And if passing null causes an error to be thrown, we might want to catch and handle it.

Reply:
I looked back at this and realized that I wasn't passing in a parameter to begin with (I was copying from the example that had a title parameter), so I think it was already passing in null. I tested it out with a title parameter and that didn't throw an error either. Since I don't have a title for my comments I'll take it out altogether.

Reply:
ah well good to know that it being null doesn't cause any issues!  Taking it out sounds good as well. As a thought for the future, it's usually good to do some sort of validation on required parameters in the server side code.